pkg/nfd-master/nfd-master.go

```diff
@@ -758,21 +801,24 @@ func (m *nfdMaster) nfdAPIUpdateAllNodes() error {
 	}
 
 	for _, node := range nodes.Items {
-		m.nodeUpdaterPool.addNode(node.Name)
+		m.updaterPool.addNode(node.Name)
 	}
 
 	return nil
 }
 
-func (m *nfdMaster) nfdAPIUpdateOneNode(cli k8sclient.Interface, node *corev1.Node) error {
-	if m.nfdController == nil || m.nfdController.featureLister == nil {
-		return nil
+// mergeNodeFeatures merges the features of the given NodeFeature objects into a single NodeFeatureSpec.
+func (m *nfdMaster) mergeNodeFeatures(nodeName, api string) (*nfdv1alpha1.NodeFeature, error) {
+	nodeFeatures := &nfdv1alpha1.NodeFeature{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nodeName,
+		},
 	}
 
-	sel := k8sLabels.SelectorFromSet(k8sLabels.Set{nfdv1alpha1.NodeFeatureObjNodeNameLabel: node.Name})
+	sel := k8sLabels.SelectorFromSet(k8sLabels.Set{nfdv1alpha1.NodeFeatureObjNodeNameLabel: nodeName})
 	objs, err := m.nfdController.featureLister.List(sel)
 	if err != nil {
-		return fmt.Errorf("failed to get NodeFeature resources for node %q: %w", node.Name, err)
+		return nodeFeatures, fmt.Errorf("failed to get NodeFeature resources for node %q: %w", nodeName, err)
```

Comment:
We should also somehow distinguish nodes that do not have any nodefeatures so that we don't try to process nodes that don't have any features discovered. E.g. by returning `nil` or so. WDYT?